src/components/tooltip/tooltip.js

```diff
@@ -102,12 +102,20 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $interpolate,
       }
     }
 
-    function addAriaLabel(override) {
-      if (override || !parent.attr('aria-label')) {
-        // Only interpolate the text from the HTML element because otherwise the custom text
-        // could be interpolated twice and cause XSS violations.
-        var interpolatedText = override || $interpolate(element.text().trim())(parent.scope);
+    function addAriaLabel(labelText) {
+      // Only interpolate the text from the HTML element because otherwise the custom text could
+      // be interpolated twice and cause XSS violations.
+      var interpolatedText = labelText || $interpolate(element.text().trim())(parent.scope);
+
+      // Only add the `aria-label` to the parent if there isn't already one, or if the previous
+      // `aria-label` was added by the tooltip directive.
+      if (!parent.attr('aria-label') || parent.attr('aria-labelledby') === 'tooltip') {
         parent.attr('aria-label', interpolatedText);
+
+        // Set the `aria-labelledby` attribute if it has not already been set.
+        if (!parent.attr('aria-labelledby')) {
+          parent.attr('aria-labelledby', 'tooltip');
```

Comment:
I thought aria-labelledby took the id of the element which does the labeling. So you should also be adding an id somewhere to the component (unless I'm missing one already added).

"Tooltip" seems too generic to be an id because then only one tooltip can exist on the page, and it's not prefixed with md- so could conflict with elements already on the page. I think using the panel id could work since those are generated each time, but I'd consider the prefix as well, similar to class names and component names.